pkg/messaging/sns.go

```diff
@@ -6,47 +6,47 @@ import (
 	_ "github.com/aws/aws-sdk-go/aws/credentials/stscreds"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/sns"
+	"strings"
 )
 
-// NewSNS initializes a new AWS SNS client
-func NewSNS(config *Config) (*sns.SNS, error) {
+// NewSNS initializes a new AWS SNS client and returns topicArn - default for EmailTopic topic
+func NewSNS(config *Config, params ...string) (*sns.SNS, string, error) {
 	c, err := mergeAndPopulateConfig(config)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	awscfg := &aws.Config{
 		Region:                        aws.String(c.AWSRegion),
 		CredentialsChainVerboseErrors: aws.Bool(true),
 	}
-	// if c.RoleArn != "" {
-	// 	creds := stscreds.NewCredentials(sess, c.RoleArn)
-	// 	awscfg.Credentials = creds
-	// } else if c.AccessKeyID != "" && c.SecretAccessKey != "" {
-	// 	credVal := credentials.Value{
-	// 		AccessKeyID:     c.AccessKeyID,
-	// 		SecretAccessKey: c.SecretAccessKey,
-	// 	}
-	// 	cred := credentials.NewStaticCredentialsFromCreds(credVal)
-	// 	awscfg.Credentials = cred
-	// 	sess = session.Must(session.NewSessionWithOptions(session.Options{
-	// 		Config: aws.Config{
-	// 			Credentials: cred,
-	// 			Region:      aws.String(c.AWSRegion),
-	// 		},
-	// 		SharedConfigState: session.SharedConfigEnable,
-	// 	}))
-	// }
 
 	sess, err := session.NewSession(awscfg)
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	var client *sns.SNS
 	client = sns.New(sess)
 	if client == nil {
-		return nil, err
+		return nil, "", err
 	}
-	return client, nil
+
+	topic := "EmailTopic"
```

Comment:
this is too service specific, it shouldn't be in this package

Reply:
Sure. This is also generic.  default for EmailTopic if the package user does not pass in anything in 'params ...string' or any SNS topic such as 'AnySNSTopic' in 'params ...string' - // NewSNS initializes a new AWS SNS client and returns topicArn - default for EmailTopic topic